test/ec_driver.cpp

```diff
@@ -137,39 +153,59 @@ bool repair_data_files(
 
     // re-read data
     for (unsigned i = 0; i < fec->n_data; i++) {
-        snprintf(filename, sizeof(filename), "%s.d%d", prefix, i);
+        snprintf(filename, sizeof(filename), "%s.d%0*d", prefix, zpad, i);
         if (vflag)
             std::cerr << "repair: checking data " << filename << "\n";
         if (-1 == access(filename, F_OK)) {
             if (vflag)
                 std::cerr << filename << " is missing\n";
             d_files[i] = nullptr;
-            r_files[i] = new std::ofstream(filename);
+            try {
+              r_files[i] = new std::ofstream(filename);
+            } catch (const std::ifstream::failure& e) {
+              std::cerr << "Exception opening " << filename << "\n";
+              exit(1);
+            }
         } else {
             r_files[i] = nullptr;
-            d_files[i] = new std::ifstream(filename);
+            try {
+              d_files[i] = new std::ifstream(filename);
+            } catch (const std::ifstream::failure& e) {
+              std::cerr << "Exception opening " << filename << "\n";
+              exit(1);
+            }
         }
     }
 
     for (unsigned i = 0; i < fec->n_outputs; i++) {
-        snprintf(filename, sizeof(filename), "%s.c%d", prefix, i);
+        snprintf(filename, sizeof(filename), "%s.c%0*d", prefix, zpad, i);
         if (vflag)
             std::cerr << "repair: checking coding " << filename << "\n";
         if (access(filename, F_OK)) {
             if (vflag)
                 std::cerr << filename << " is missing\n";
             c_files[i] = nullptr;
         } else {
+          try {
             c_files[i] = new std::ifstream(filename);
+          } catch (const std::ifstream::failure& e) {
```

Comment:
We really have a lot of duplicated code/copy-paste in this file (all the loop to open streams), this could probably be factored in an helper function.

A simple change like that shouldn't require so many modifications :/